### PR TITLE
MWPW-142249: checkout-link flows

### PR DESCRIPTION
**Pull request by yesil**
for the following use cases:

1. download
2. upgrade
3. twp modal
4. d2p modal

Includes: https://git.corp.adobe.com/wcms/tacocat.js/pull/535

Resolves: [MWPW-142249](https://jira.corp.adobe.com/browse/MWPW-142249)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/ilyas/checkout-links?martech=off
- After:  https://mwpw-142249--milo--yesil.hlx.page/drafts/ilyas/checkout-links?martech=off

**Test with iMS:** 
https://ims--milo--adobecom.hlx.page/drafts/ilyas/checkout-links?martech=off

**CC Test URLs:**
https://main--cc--adobecom.hlx.page/drafts/ilyas/checkout-links?milolibs=mwpw-142249--milo--yesil

**Screenshots:**

IMG_0386

IMG_0388

IMG_0387

### CODE REVIEW

**Review by VKniaz — APPROVED** [review]

**Review by 3ch023 — APPROVED** [review]
Verified Switch Modal as well
https://main--cc--adobecom.hlx.page/cc-shared/fragments/drafts/commerce/testpage?milolibs=MWPW-142249--milo--yesil

File: libs/blocks/global-navigation/utilities/getUserEntitlements.js [hunk 1]
```diff
@@ -94,7 +95,7 @@ const mapSubscriptionCodes = (allOffers) => {
   };
 };
 
-const getSubscriptions = async ({ queryParams }) => {
+const getSubscriptions = async ({ queryParams }, format) => {
```
Comment (mokimo) [inline libs/blocks/global-navigation/utilities/getUserEntitlements.js:98]:
```suggestion
const getSubscriptions = async ({ queryParams, format }) => {
```

File: libs/blocks/global-navigation/utilities/getUserEntitlements.js [hunk 2]
```diff
@@ -125,30 +126,30 @@ const getSubscriptions = async ({ queryParams }) => {
...
       : entitlements[queryParams]
         .then((res) => mapSubscriptionCodes(res))
-        .catch(() => emptyEntitlements());
+        .catch(() => emptyEntitlements(format));
   }
 
   entitlements[queryParams] = entitlements[queryParams]
    || new Promise((resolve) => {
-     getSubscriptions({ queryParams })
+     getSubscriptions({ queryParams }, format)
```
Comment (mokimo) [inline libs/blocks/global-navigation/utilities/getUserEntitlements.js:143]:
```suggestion
     getSubscriptions({ queryParams, format })
```

**Review by mokimo — APPROVED** [review]
I have not checked the merch related code in detail, but from a global-navigation/getUserEntitlements code owner perspective that looks good with a small suggestion! 👍

File: libs/blocks/global-navigation/utilities/getUserEntitlements.js (hunk 2 above)
Comment (yesil) [reply libs/blocks/global-navigation/utilities/getUserEntitlements.js:143]: @mokimo  `getSubscriptions` expects two params, I didn't want to change its signature.

**Pull request merged**